iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -55,16 +55,74 @@ static bool isAllZero(DenseIntElementsAttr attr) {
   return attr.getSplatValue<IntegerAttr>().getInt() == 0;
 }
 
-/// Returns true if the linalg op has padding attribute, and that it has
+static bool isIota(ArrayRef<int64_t> array) {
+  for (auto it : llvm::enumerate(array)) {
+    if (it.index() != it.value()) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/// Returns true if the conv op has padding attribute, and that it has
 /// non-zero entries.
-template <typename OpTy>
-static bool hasPadding(OpTy op) {
+static bool hasPadding(mhlo::ConvOp op) {
   Optional<DenseIntElementsAttr> padding = op.padding();
   if (!padding) return false;
   return llvm::any_of(padding.getValue(),
                       [](APInt v) -> bool { return !v.isNullValue(); });
 }
 
+/// Returns true if the conv op has stride attribute, and that it has
+/// non-zero entries.
+static bool hasStride(mhlo::ConvOp op) {
+  Optional<DenseIntElementsAttr> stride = op.window_strides();
+  if (!stride) return false;
+  return llvm::any_of(stride.getValue(),
+                      [](APInt v) -> bool { return !v.isOneValue(); });
+}
+
+/// Returns true if the conv op has input dilation.
+static bool hasInputDilation(mhlo::ConvOp op) {
+  Optional<DenseIntElementsAttr> dilation = op.lhs_dilation();
+  if (!dilation) return false;
+  return llvm::any_of(dilation.getValue(),
+                      [](APInt v) -> bool { return !v.isOneValue(); });
+}
+
+/// Returns true if the conv op has kernel dilation.
+static bool hasKernelDilation(mhlo::ConvOp op) {
+  Optional<DenseIntElementsAttr> dilation = op.rhs_dilation();
+  if (!dilation) return false;
+  return llvm::any_of(dilation.getValue(),
+                      [](APInt v) -> bool { return !v.isOneValue(); });
+}
+
+static bool hasNormalizedConvolution(mhlo::ConvOp op) {
+  auto dimensionNumbers = op.dimension_numbers();
+
+  if (dimensionNumbers.input_batch_dimension().getValue().getSExtValue() != 0)
+    return false;
+  if (dimensionNumbers.input_batch_dimension().getValue().getSExtValue() != 0)
+    return false;
```

Comment:
nit: I'd prefer to always include braces for these statements, but we have mixed style...

Reply:
Done.